keep/api/bl/ai_suggestion_bl.py

```diff
@@ -0,0 +1,499 @@
+import hashlib
+import json
+import logging
+import uuid
+from typing import Dict, List, Optional, Set, Tuple
+from uuid import UUID
+
+from fastapi import HTTPException
+from openai import OpenAI, OpenAIError
+from sqlmodel import Session
+
+from keep.api.bl.incidents_bl import IncidentBl
+from keep.api.core.db import get_session_sync
+from keep.api.models.alert import (
+    AlertDto,
+    IncidentCandidate,
+    IncidentClustering,
+    IncidentDto,
+    IncidentsClusteringSuggestion,
+)
+from keep.api.models.db.ai_suggestion import AIFeedback, AISuggestion, AISuggestionType
+from keep.api.models.db.topology import TopologyServiceDtoOut
+
+
+class AISuggestionBl:
+    def __init__(self, tenant_id: str, session: Session | None = None) -> None:
+        self.logger = logging.getLogger(__name__)
+        self.tenant_id = tenant_id
+        self.session = session if session else get_session_sync()
+
+        # Todo: interface it with any model
```

Comment:
Can we add issues and mention links for TODOs?

Reply:
done